client/src/index.ts

```diff
@@ -90,19 +90,78 @@ type FilterConstraints<T> = {
   [key in keyof T]?: T[key] extends Record<string, any> ? FilterConstraints<T[key]> : T[key] | DeepConstraint<T[key]>;
 };
 
+type CursorNavigationOptions = { first?: string } | { last?: string } | { after?: string; before?: string };
+type OffsetNavigationOptions = { size?: number; offset?: number };
+type PaginationOptions = CursorNavigationOptions & OffsetNavigationOptions;
+
 type BulkQueryOptions<T> = {
+  page?: PaginationOptions;
+  /** TODO: Not implemented yet
   filter?: FilterConstraints<T>;
   sort?:
     | {
         column: keyof T;
         direction?: SortDirection;
       }
     | keyof T;
+**/
```

Comment:
Added here the options we don't have implemented. Commented so that they do not autocomplete.